protos/format.proto

```diff
@@ -43,6 +43,9 @@ Format:
 message Manifest {
   // All fields of the dataset, including the nested fields.
   repeated Field fields = 1;
+
+  // Dataset metadata.
+  map<string, bytes> metadata = 2;
```

Comment:
wondering why Arrow has the metadata keys as bytes as well. Is it that they forgot to convert some type or people are putting non-string as keys? I'm guessing it's for non-unicode keys maybe?

Reply:
It is strings in arrow's C++ codebase

https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/key_value_metadata.h#L79-L80